### PR TITLE
Final Group Capstone - Book a Hotel Room - Back-end

### PR DESCRIPTION
**Pull request by anyars-encarta**
**Hi Reviewer 👋**

In this project, we created a rails API app that communicates to a [react-redux front-end](https://github.com/anyars-encarta/hotel-booking-front-end/pull/29). 

The details of the project are below:
- [ ] We created the following models:

   - User
   - Category
   - Room
   - Reservation
 
- [ ] We created the following API controllers:

   - users_controller
   - categories_controller
   - rooms_controller
   - reservations_controller

- [ ] We created routes to provide all the necessary API end-points for react front-end app.
- [ ] We also configured the CORS to allows cross origin access to the backed.
- [ ] We added full documentation for our API.
- [ ] We added unit tests for the controllers using **RSpec**

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# This file should ensure the existence of records required to run the application in every environment (production,
+# development, test). The code here should be idempotent so that it can be executed at any point in every environment.
+# The data can then be loaded with the bin/rails db:seed command (or created alongside the database with db:setup).
+#
+# Example:
+#
+#   ["Action", "Comedy", "Drama", "Horror"].each do |genre_name|
+#     MovieGenre.find_or_create_by!(name: genre_name)
+#   end
+# Users
```
Comment (vichuge) [inline db/seeds.rb:10]: - Creating seeds could be an excellent way to improve the project. 🐴

**Review by vichuge — COMMENTED** [review]
Hi dear student, 

Good job so far!
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!

mario_ladders


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

- It looks like there are some failures in the tests, I kindly invite you to solve. 🐼 

```
Finished in 0.49367 seconds (files took 1.06 seconds to load)
54 examples, 10 failures

Failed examples:

rspec ./spec/controllers/categories_controller_spec.rb:26 # Api::CategoriesController GET #show returns a successful response
rspec ./spec/controllers/categories_controller_spec.rb:33 # Api::CategoriesController GET #show returns the category in JSON format
rspec ./spec/controllers/reservations_controller_spec.rb:10 # Api::ReservationsController GET #index returns all reservations in JSON format
rspec ./spec/controllers/reservations_controller_spec.rb:29 # Api::ReservationsController POST #create with valid parameters creates a new reservation
rspec ./spec/controllers/reservations_controller_spec.rb:48 # Api::ReservationsController Un reserved room un reserves a room
rspec ./spec/controllers/rooms_controller_spec.rb:10 # Api::RoomsController GET #index returns all rooms in JSON format
rspec ./spec/controllers/rooms_controller_spec.rb:27 # Api::RoomsController GET #show returns a successful response
rspec ./spec/controllers/rooms_controller_spec.rb:35 # Api::RoomsController GET #show returns the room in JSON format
rspec ./spec/controllers/rooms_controller_spec.rb:49 # Api::RoomsController POST #create creates a new room
rspec ./spec/controllers/rooms_controller_spec.rb:67 # Api::RoomsController DELETE #destroy deletes a room

```
### Front-end CR [here](https://github.com/anyars-encarta/hotel-booking-front-end/pull/29)

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: db/seeds.rb (hunk 1 above)
Comment (MozamelJawad) [reply db/seeds.rb:10]: Thanks for the review, Dear @vichuge 

- Seeds have been created.
- We have checked the tests and it's all working on our side.
image

Comment (anyars-encarta) [reply db/seeds.rb:10]: @vichuge , We very much appreciate your very detailed review. However, we are confused on the fact that all the tests are passing at our end while failing at your end. Its making it difficult for us to be able to identify where the issue might be.

We have also included the seed to improve the app as you suggested.

Thank you.

**Review by Meltrust — APPROVED** [review]
# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight


- Superb job ✔️  



Cheers and Happy coding!👏👏👏


------

**Pull request merged**